twittermap/public/javascripts/map/controllers.js

```diff
@@ -90,6 +90,13 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common'])
       });
     }
 
+    function resetGeoInfo(level) {
+      $scope.status.logicLevel = level;
+      if ($scope.geojsonData[level])
+        resetGeoIds($scope.bounds, $scope.geojsonData[level], level + 'ID');
+      cloudberry.parameters.geoLevel = level;
```

Comment:
a picky comment, 

it feels very unnature to have two lines before and after the `if` clause. I was thinking `cloudberry.parameters.geoLevel = level;` is inside the if. 

It will be better to group the non-branch statements together. 
```
   $scope.status.logicLevel = level;
   cloudberry.parameters.geoLevel = level;
   if ($scope.geojsonData[level])
       resetGeoIds($scope.bounds, $scope.geojsonData[level], level + 'ID');
```

or put it all after the if. you decide :-)